### ISSUE TITLE
Improve resolution error message when dependency is locked

### ISSUE THREAD

**Issue by ericmj** [hexpm/hex#109]
Error message:

```
Looking up alternatives for conflicting requirements on postgrex
  Activated version: 0.8.1
  From deps/ecto/mix.exs: ~> 0.9.1
  From deps/hex_web/mix.exs: >= 0.0.0
```

The resolution failed because postgrex was locked to 0.8.1. We should improve the error message so this is clear.

**Comment by montague** [comment]
i'll take a crack at this

**Comment by montague** [comment]
is this as simple as appending "try running `mix hex.unlock [pkg]`"? i might be looking too deeply into this if that's the case.

also, the message indicates that we're looking for a solution to the problem--is that accurate?

i've been looking in `lib/hex/resolver.ex`, figured i'd double-check with you.

**Comment by ericmj** [comment]
You are on the right track. Today we just say `Activated version: 0.8.1` to say which version we tried and failed with, then we list the requirements that applies to the version (some of which fail to match the activated version). The confusing thing is why we activated version `0.8.1` when the requirements are `~> 0.9.1` and `>= 0.0.0`. The version was activated before resolution began because it came from the lock. We should add the mix.lock version to the list of requirements we print, adding `From mix.lock: 0.8.1` to the list should do it.

You are right that we are actively looking for a solution, we effectively try every combination of package versions (with a lot of optimizations and early reductions) and each time we "failed" with a set of activated versions we have to backtrack and try a new set, each time we backtrack we print the above message to give the user as much information as possible why the resolution failed.
